src/libraries/System.Private.CoreLib/src/System/Collections/Generic/IDictionaryDebugView.cs

```diff
@@ -11,9 +11,7 @@ internal sealed class IDictionaryDebugView<TKey, TValue> where TKey : notnull
 
         public IDictionaryDebugView(IDictionary<TKey, TValue> dictionary)
         {
-            ArgumentNullException.ThrowIfNull(dictionary);
-
-            _dict = dictionary;
+            _dict = dictionary ?? throw new ArgumentNullException(nameof(dictionary));
```

Comment:
System.Collections.Immutable still targets old framework and the throw method is unavailable.
The performance difference shouldn't matter as this is supposed as debugger-only.